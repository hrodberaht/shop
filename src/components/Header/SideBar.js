import React, { Component } from 'react';
import { NavLink, Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { connect } from 'react-redux';
import ConnectedAuthorization from '../Auth/Authorization';
import { getNavigationSidebarValue } from '../../store/navigation/selectors';

export class SideBar extends Component {
  render() {
    const { toggleSidebar } = this.props;
    return (
      <div className={classNames({ sidebarHide: !toggleSidebar, sidebar: toggleSidebar })}>
        <ul className="sidebar__list">
          <li className="item">
            <Link to="/">Logo</Link>
          </li>
          <hr />
          <li className="item">
            <NavLink exact to="/">
              Shop
            </NavLink>
          </li>
          <li className="item">
            <NavLink to="/orders">Orders</NavLink>
          </li>
          <li className="item">
            <NavLink to="/cart">Cart</NavLink>
          </li>
          <li className="item">
            <NavLink to="/wishlist">Wish list</NavLink>
          </li>
          <ConnectedAuthorization
            render
            withRoleAdmin={(
              <li className="item">
                <NavLink to="/inventory">Inventory</NavLink>
              </li>
)}
          />
          <ConnectedAuthorization
            logout={logout => (
              <li className="item">
                <Link to="/" onClick={logout}>
                  Logout
                </Link>
              </li>
            )}
          />
        </ul>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  toggleSidebar: getNavigationSidebarValue(state),
});

export default withRouter(connect(mapStateToProps)(SideBar));

SideBar.propTypes = {
  toggleSidebar: PropTypes.bool.isRequired,
};
