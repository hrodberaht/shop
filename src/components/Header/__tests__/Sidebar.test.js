import { SideBar } from '../SideBar';

const defaultProps = {
  toggleSidebar: true,
};
const setup = buildSetup(SideBar, defaultProps);

describe('<SideBar />', () => {
  it('should render with side bar if toggleSidebar is true', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with hide sidebar if toggleSidebar is false', () => {
    const { wrapper } = setup();
    wrapper.setProps({ toggleSidebar: false });
    expect(wrapper).toMatchSnapshot();
  });
});
