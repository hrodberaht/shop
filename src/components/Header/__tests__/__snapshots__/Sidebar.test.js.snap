// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> should render with hide sidebar if toggleSidebar is false 1`] = `
<div
  className="sidebarHide"
>
  <ul
    className="sidebar__list"
  >
    <li
      className="item"
    >
      <Link
        replace={false}
        to="/"
      >
        Logo
      </Link>
    </li>
    <hr />
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        exact={true}
        to="/"
      >
        Shop
      </NavLink>
    </li>
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/orders"
      >
        Orders
      </NavLink>
    </li>
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/cart"
      >
        Cart
      </NavLink>
    </li>
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/wishlist"
      >
        Wish list
      </NavLink>
    </li>
    <Connect(Authorization)
      render={true}
      withRoleAdmin={
        <li
          className="item"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/inventory"
          >
            Inventory
          </NavLink>
        </li>
      }
    />
    <Connect(Authorization)
      logout={[Function]}
    />
  </ul>
</div>
`;

exports[`<SideBar /> should render with side bar if toggleSidebar is true 1`] = `
<div
  className="sidebarHide"
>
  <ul
    className="sidebar__list"
  >
    <li
      className="item"
    >
      <Link
        replace={false}
        to="/"
      >
        Logo
      </Link>
    </li>
    <hr />
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        exact={true}
        to="/"
      >
        Shop
      </NavLink>
    </li>
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/orders"
      >
        Orders
      </NavLink>
    </li>
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/cart"
      >
        Cart
      </NavLink>
    </li>
    <li
      className="item"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/wishlist"
      >
        Wish list
      </NavLink>
    </li>
    <Connect(Authorization)
      render={true}
      withRoleAdmin={
        <li
          className="item"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/inventory"
          >
            Inventory
          </NavLink>
        </li>
      }
    />
    <Connect(Authorization)
      logout={[Function]}
    />
  </ul>
</div>
`;
